### ISSUE TITLE
Investigate showing diagnostics panel on save

### ISSUE THREAD

**Issue by tomv564** [tomv564/LSP#381]
Instead of the debounced delay logic in #292

**Comment by predragnikolic** [comment]
Having an option to show the panel on save would be cool. But there is one other thing that needs improvement.

This plugin currently handles diagnostics on this way:
* Open a file, diagnostics are shown for that file.
* Close a file, diagnostics are removed for that file.

I think this is a better approach.
* Open a file, diagnostics are shown for that file.
* Close a file, diagnostics are shown for that file.

I thought about when should the diagnostics be removed, and it is only in this two cases:
* When a file is opened and it doesn't belong to the project. As soon as it is closed, the diagnostics for that file should be removed.
* When a file is opened in a project, but it belongs to a vendor(PHP), node_module(JavaScript) or some other folder that should be ignored. When the file is closed all diagnostics should be removed, we don't want to care for code that we didn't write. This would need a new setting, like `ignore_diagnostics_paths` which would take a list of folder names to ignore. like ['vendor', 'node_modules'].

The advantage to this is when you close a file and it had errors, you would still see those error. Before you wouldn't know because as soon as you close it, all diagnostics for the given file are removed. 

I already made like a prototype(I hardcoded the ignored paths). if you want you can try it. 
In `core/windows.py` go to line 449 and replace the contents of `_handle_view_closed` with this:
```python
 def _handle_view_closed(self, view: View, session):
        file_name = view.file_name()
        base_dir = get_project_path(self._window)

        if not base_dir or not file_name:
            return

        file_in_project = base_dir in file_name  # type: bool
        file_in_ignored_paths = self._is_in_ignored_paths(base_dir, file_name)

        if not file_in_project or file_in_ignored_paths:
            self._diagnostics.remove(view, session.config.name)
            self._sublime.set_timeout_async(lambda: self._check_window_closed(), 500)

    def _is_in_ignored_paths(self, base_dir, file_name) -> bool:
        """ Returns a bolean indicating if the file should be ignored """
        ignored_paths = ['stubs', 'tests']  # this will be populated by the settings
        for ignored_path in map(lambda path: os.path.join(base_dir, path), ignored_paths):
            if ignored_path in file_name:
                return True
        return False
```
So now when you open a file in the LSP project or any Project that has `stubs` or `tests` at the root, as soon as you close the file that are in these folders the diagnostics will go away. Same for files that you open but don't belong to the currently open project. Only errors that matter to you will stay visible. 

Sorry for all the details, I wanted to be clear.

**Comment by tomv564** [comment]
I mirrored the behaviour of VS Code in this implementation, even though I personally also like to see more errors rather than less.

As long as a file is not open in LSP, we don't really know if the errors are still valid (the language server won't tell us when a closed file is fixed). So if you fix a problem in file A that solves the problem in file B, you are stuck with the diagnostics for file B until you re-open it.

Also, `ignore_diagnostics_paths` would be a lot of effort to set up for each language and project :).

**Comment by tomv564** [comment]
Closed by #754, you can set `auto_show_diagnostics_panel` to `"saved"` to get it to open when diagnostics exist while a file is saved.
